include/uam/uam_driver_ros.h

```diff
@@ -152,8 +148,51 @@ class UamROS
 
     msg.header.stamp = wall_time + time_offset + ros::Duration(scan_params_.getAngularTimeOffset());
 
-    // Read the right fields
-    fillScanMessageData(reply, range_offset, msg);
+    // First and last steps
+    auto first_step = scan_params_.getFirstStep();
+    auto last_step = scan_params_.getLastStep();
+
+    const auto number_of_readings = last_step - first_step + 1;
+
+    if (reply.ranges.size() < number_of_readings)
+    {
+      ROS_ERROR_STREAM(
+        "Unexpected outcome: " << reply.ranges.size() << ", first and last_step are " << first_step << ", "
+                               << last_step);
+      return;
+    }
```

Comment:
In the original code, you had the check:
```
  if (reply.ranges.size() <= last_step)
  {
    ROS_ERROR_STREAM("Unexpected outcome: " << reply.ranges.size() << " and last_step is " << last_step);
    return;
  }
```
That check seems more correct to me. The `reply.ranges` container appears to be the source container that has all of the range readings, not just the requested subset. Therefore the size of `reply.ranges` must be greater than `last_step` and not merely greater than the size of the requested subset.